src/item.cpp

```diff
@@ -7408,7 +7413,26 @@ int item::lift_strength() const
 
 int item::attack_time( const Character &you ) const
 {
-    int ret = 65 + ( volume() / 62.5_ml + weight() / 60_gram ) / count();
+    const double base_move = 65;
+    const double const_stab = 3;
+    const double const_weild = 2.5;
+    const double sword_factor = 0.75;
+
+    int ret = base_move;
+    int stab_damage = damage_melee( damage_stab );
+    if( stab_damage > damage_melee( damage_bash ) && stab_damage > damage_melee( damage_cut ) ) {
```

Comment:
see my comment on the issue. This part in particular should preferentially be using to-hit data rather than damage types, and should default to damage types only if the to-hit data is empty.